docs/rfcs/0002_templates_mvp.md

```diff
@@ -0,0 +1,339 @@
+
+-   Any organization or study administrator may update a Template or
+    TemplateVersion if their organization owns it
+
+## 🗑 Deleting Templates
+
+### DeleteTemplateMutation
+
+```graphql
+type DeleteTemplateInput {
+    id: ID!
+}
+```
+
+### DeleteTemplateVersionMutation
+
+```graphql
+type DeleteTemplateVersionInput {
+    id: ID!
+}
+```
+
+### Permissions
+
+### Permissions
+
+-   Any organization or study administrator may delete a Template or
+    TemplateVersion if their organization owns it
+
+## 📦 Downloading Templates
+
+### Download Study Templates
+
+Probably the easiest way to download templates since it bundles everything
+into an archive. You will get an archive containing:
+
+For each TemplateVersion in the study:
+
+-   TemplateVersion.submission_template_file: the TSV file that people will fill out and use for data submission to their study
+-   TemplateVersion.fields_schema: the template field definitions which serve as instructions on filling out the template
+
+    /download/study/<Study.kf_id>templates
+
+We should consider providing a multi-sheet Excel workbook with a sheet for each
+template's fields schema and a folder of TSVs with all of the submission
+templates.
+
+### Download Latest Template
+
+Download the latest version of template.
+
+```
+/download/template/<Template.id>
+```
+
+You would get an archive with two files:
+
+-   Template.latest_version.submission_template_file
+-   Template.latest_version.fields_schema
+
+### Download a Template Version
+
+Download a specific version of a template.
+
+```
+/download/template/<Template.id>/versions/<TemplateVersion.id>
+```
+
+You would get an archive with two files:
+
+-   TemplateVersion.submission_template_file
+-   TemplateVersion.fields_schema
+
+### Permissions
+
+-   Any user may download template files
+
+## Changes to Existing Functionality
+
+-   A `Study` will have a new field, `Study.template_versions`, a many to many
+    relationship to `TemplateVersion`.
+-   The `File.file_type` field will be replaced by a one to one foreign key
+    to a `TemplateVersion`. The `TemplateVersion` must be one of
+    templates in the Study's template versions.
+-   The study and file mutations should be updated accordingly as well
+
+## 💻 Implementation Considerations
+
+When thinking about implementing data submission templates, we should consider
+the entire development, test, and documentation lifecycle and some of the
+common painpoints encountered as they may inform a better implementation of our
+templates.
+
+### Single Source of Truth
+
+A change in my template definition (e.g. new column) should not cause a
+cascading set of manual changes that need to be made in several other
+related artifacts.
+
+For example, if I add a new column to the Participant Diseases template,
+I need to ensure that the following are also updated to reflect the change:
+
+-   Template documentation / end-user guidance
+-   Test data generators in template unit tests
```

Comment:
If we are updating our tests to account for all real-world templates, it's not a unit test. Unit testing would be around features of templates where we would use templates that clearly demonstrate desired behavior with the minimal amount of definition.

Reply:
Yea I agree. I just think it would be nice if we could define the templates and then somehow generate data that conforms to the fields schema automatically so that it could easily be used in testing. It will also make testing extract configs easier (more on this later).